autoapi/mappers/dotnet.py

```diff
@@ -133,33 +147,32 @@ def _recurse_ns(obj):
             if len(ns.children) == 0:
                 del self.namespaces[key]
 
-    # def output_rst(self, root, source_suffix):
-    #     for obj in self.top_level_objects.values():
+    def output_rst(self, root, source_suffix):
+        for id, obj in self.objects.items():
 
-    #         if not obj:
-    #             continue
+            if not obj or not obj.top_level_object:
+                continue
 
-    #         rst = obj.render()
-    #         if not rst:
-    #             continue
+            rst = obj.render()
+            if not rst:
+                continue
 
-    #         # Detail
-    #         try:
-    #             filename = obj.name.split('(')[0]
-    #         except IndexError:
-    #             filename = obj.name
-    #         detail_dir = os.path.join(root, *filename.split('.'))
-    #         ensuredir(detail_dir)
-    #         # TODO: Better way to determine suffix?
-    #         path = os.path.join(detail_dir, '%s%s' % ('index', source_suffix))
-    #         with open(path, 'w+') as detail_file:
-    #             detail_file.write(rst.encode('utf-8'))
-
-    #     # Write Indexes
-    #     top_level_index = os.path.join(root, 'index.rst')
-    #     with open(top_level_index, 'w+') as top_level_file:
-    #         content = self.jinja_env.get_template('index.rst')
-    #         top_level_file.write(content.render(pages=self.namespaces.values()))
+            try:
+                filename = obj.name.split('(')[0]
+            except IndexError:
+                filename = id
+            filename = filename.replace('#', '-')
+            detail_dir = os.path.join(root, *filename.split('.'))
+            ensuredir(detail_dir)
+            path = os.path.join(detail_dir, '%s%s' % ('index', source_suffix))
+            with open(path, 'w+') as detail_file:
+                detail_file.write(rst.encode('utf-8'))
+
+        # Render Top Index
+        top_level_index = os.path.join(root, 'index.rst')
+        with open(top_level_index, 'w+') as top_level_file:
+            content = self.jinja_env.get_template('index.rst')
+            top_level_file.write(content.render(pages=self.namespaces.values()))
```

Comment:
Again here, if we need to catch and warn on this. Also could be fine to die and bubble up the error on these blocks I suppose.